src/main/webapp/script.js

```diff
@@ -46,23 +32,44 @@ function createPostElement(post) {
     const postElement = document.createElement('li');
     postElement.className = 'post';
     
-    const titleElement = document.createElement('span');
-    titleElement.innerText = post.textValue;
-    
-    // create a button to delete an elemnet
-    const deleteButtonElement = document.createElement('button');
-    deleteButtonElement.innerText = 'Delete';
-    deleteButtonElement.addEventListener('click', () => {
-        delete post;
-    // Remove an element from the DOM.
-    postElement.remove();
-    });
-    
-    postElement.appendChild(titleElement);
-    postElement.appendChild(deleteButtonElement);
+    const nameElement = document.createElement('strong');
+    nameElement.innerText = "Pet Name: " + post.petName;
+
+    const typeElement = document.createElement('div');
+    typeElement.innerText = "Animal Type: " + post.animalType;
+
+    const locElement = document.createElement('div');
+    locElement.innerText = "Location: " + post.location;
+
+    const ageElement = document.createElement('div');
+    var dob = new Date(post.dob.year, post.dob.month, post.dob.day);
+    ageElement.innerText = "Age: "+ getAge(dob) + " years old";
+
+    postElement.appendChild(nameElement);
+    postElement.appendChild(typeElement);
+    postElement.appendChild(locElement);
+    postElement.appendChild(ageElement);
+
     return postElement;
 }
 
+/** Caculate the age based on birthday. (minus for unborn baby)*/
+function getAge(dateString) 
+{
+    var today = new Date(); //returns date object of current date and time
+    var birthDate = new Date(dateString);
+    var age = today.getFullYear() - birthDate.getFullYear();
+    var m_diff = today.getMonth() - birthDate.getMonth();
+    if (m_diff < 0) 
+    {
+        age--;
+    }
+    
+    return age;
```

Comment:
it might make it more complicated but it would be nice to see the age in months if it is younger than a year. You could also just always show age in years and months